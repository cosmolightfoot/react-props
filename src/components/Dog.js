import React from 'react';
import PropTypes from 'prop-types';

export default function Dog({ name, age, weight }) {
  return (
    <dl>
      <dt>Name</dt>
      <dd>{name}</dd>

      <dt>Age</dt>
      <dd>{age}</dd>

      <dt>Weight</dt>
      <dd>{weight}</dd>
    </dl>
  );

}
Dog.propTypes = {
  name: PropTypes.string.isRequired,
  age: PropTypes.number.isRequired,
  weight: PropTypes.string
};
