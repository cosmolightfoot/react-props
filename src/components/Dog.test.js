import React from 'react';
import { shallow } from 'enzyme';
import Dog from './Dog';
import { exportAllDeclaration } from '@babel/types';

describe('Dog component', () => {
  it('renders a dog', () => {
    const wrapper = shallow(<Dog name="spot" age={5} weight="80kg" />);
    expect(wrapper).toMatchSnapshot();
  });
});
