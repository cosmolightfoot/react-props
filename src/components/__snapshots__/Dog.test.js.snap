// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dog component renders a dog 1`] = `
<dl>
  <dt>
    Name
  </dt>
  <dd>
    spot
  </dd>
  <dt>
    Age
  </dt>
  <dd>
    5
  </dd>
  <dt>
    Weight
  </dt>
  <dd>
    80kg
  </dd>
</dl>
`;
